### DISCUSSION TITLE
LEGO Liebherr LR 13000 / Xbox Controller / Pybricks

### DISCUSSION THREAD

**Discussion by seecheck** [pybricks/support#1505] [Show and tell]
**Mod edit: This community project was using an early version that no longer works. Please use the updated [official examples](https://pybricks.com/project/lego-42146/) if you are new to Pybricks**

Years ago me and my sons we spent nice times together playing with LEGO, especially with City sets. "The times they are a-changin' ", my sons are long grown up serious men and I am a crazy old man playing with Lego Technic since the end of last year. Last week I learned about the possibility to connect Xbox controller with Lego hubs, so I decided to try to write a code for LIebherr LR 13000 / Xbox controller using Pybricks beta. I have to say that I am not a programmer and I have only very basic knowledge in this field. So I searched, tried to understand, put together ... Thanks to everyone who shares their great knowledge here. My result is probably not very perfect, but it works. Controlling the crane is much more comfortable and precise with this controller. So ...

Run the attached codes on the crane hubs.
[LR13000-XboxController.zip](https://github.com/pybricks/support/files/14485859/LR13000-XboxController.zip)

Top hub connects to Xbox controller.
Bottom hub receives the control data from the top one.
Controls:
Undercarriage
    Left track forward ... Left trigger
    Left track backward ... Left trigger + Button X pushed
    Right track forward ... Right trigger
    Right track backward ... Right trigger + Button A pushed
    The superstructure turning ... Right joystick left / right
Crane
    The boom raising / lowering ... Left joystick up / down
    The jib raising / lowering ... Left joystick left / right
    The hook raising / lowering ... Right joystick up / down		
    The lights on / off ... Button Y

Note: I have added lights on the crane boom (connected to the top hub). Please delete the relevant lines from the code if you  test it without this equipment.

**Comment by tstr14** [comment]
First and foremost, thanks for writing this making it available! I have the LR 13000 in highly modified form and was wondering if your program has limitations like the LEGO factory settings do? Because I have extended the boom, I also needed to replace the cable with longer cord. If there are no auto shut offs for either up or down on the hook, it would be fine with me. Also, I too have added lights to the unit. Where did you find a plug to fit the hub, and how is it wired? Thanks in advance for any info you can provide!

**Reply by seecheck** [reply]
There is no calibration procedure in my code so no auto shut offs for the motors. It is necessary to monitor a position of the hook, the boom and the jib visually in operation and slow and stop the motors in time.
As for the lights, I have used Powered Up light (part no. 88005) in my model, connected to the top hub port C. 
![P_20240408_191633](https://github.com/pybricks/support/assets/162053803/0f93c7f1-2b08-4aba-bcdb-66e4351d0675)
